src/test/java/StringManipulationTest.java

```diff
@@ -53,59 +70,102 @@ public void testRemoveNthCharacter2() {
+        });
     }
 
+    // test case for n < 0
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("coffee");
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedstring.removeNthCharacter(-1, true);
+        });
     }
 
+    // test case for n = 0
     @Test
-    public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+    public void testRemoveNthCharacter7() {
+        manipulatedstring.setString("coffee");
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedstring.removeNthCharacter(0, true);
+        });
     }
 
+    // test case for empty string
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("");
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedstring.removeNthCharacter(3, true);
+        });
     }
 
-    @Test
-    public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
-    }
 
     @Test
     public void testGeSubStrings1() {
         manipulatedstring.setString("This is my string");
         String [] sStings = manipulatedstring.getSubStrings(3, 4);
-
         assertEquals(sStings[0], "my");
         assertEquals(sStings[1], "string");
     }
 
+    // if start word > end word
     @Test
     public void testGeSubStrings2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is my string");
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedstring.getSubStrings(4, 2);
+        });
     }
+
+    // if start word < 0
     @Test
     public void testGeSubStrings3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is my string");
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedstring.getSubStrings(0, 2);
+        });
     }
+
+    // if end word < 0
     @Test
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is my string");
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedstring.getSubStrings(2, 0);
+        });
     }
+
+    // if end word > amount of words
     @Test
     public void testGeSubStrings5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is my string");
+        assertThrows(IndexOutOfBoundsException.class, () -> {
+            manipulatedstring.getSubStrings(2, 6);
+        });
     }
+
+    // if start and end indexes are the same
     @Test
     public void testGeSubStrings6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is my string");
+        String [] sStings = manipulatedstring.getSubStrings(3, 3);
+        assertEquals(sStings[0], "my");
+        assertEquals(sStings[1], null);
```

Comment:
The result of getSubStrings(start, end) should be (end - start + 1) so there should not be an index 1 to index here.

Reply:
I've split it into an array so if there's two words they will always be [0] and [1] at least that's how its coded